src/platform/datapath_iouring.c

```diff
@@ -0,0 +1,2264 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Environment:
+
+    Linux
+
+--*/
+
+#include "platform_internal.h"
+#include "datapath_linux.h"
+
+#ifdef QUIC_CLOG
+#include "datapath_iouring.c.clog.h"
+#endif
+
+CXPLAT_STATIC_ASSERT((SIZEOF_STRUCT_MEMBER(QUIC_BUFFER, Length) <= sizeof(size_t)), "(sizeof(QUIC_BUFFER.Length) == sizeof(size_t) must be TRUE.");
+CXPLAT_STATIC_ASSERT((SIZEOF_STRUCT_MEMBER(QUIC_BUFFER, Buffer) == sizeof(void*)), "(sizeof(QUIC_BUFFER.Buffer) == sizeof(void*) must be TRUE.");
+
+//
+// Context value within the IoSqe to indicate the type of IO operation.
+//
+typedef enum DATAPATH_CONTEXT_TYPE {
+    DatapathContextRecv,
+    DatapathContextSend,
+} DATAPATH_CONTEXT_TYPE;
+
+//
+// Contains all the info for a single RX IO operation. Multiple RX packets may
+// come from a single IO operation.
+//
+typedef struct __attribute__((aligned(16))) DATAPATH_RX_IO_BLOCK {
```

Comment:
didn't you define a handy constant, `CXPLAT_MEMORY_ALIGNMENT` in another header?